File: src/com/leetcode/Easy/BuildAnArrayWithStackOperations.java
package com.leetcode.Easy;

import java.util.*;

// 1441. Build an Array With Stack Operations
public class BuildAnArrayWithStackOperations {
    public List<String> buildArray(int[] target, int n) {
        List<String> result = new ArrayList<>();
        int i = 1, j = 0;
        while (i <= n) {
            if (j == target.length)
                break;

            if (i == target[j]) {
                result.add("Push");
                j++;
            } else {
                result.add("Push");
                result.add("Pop");
            }
            i++;
        }
        return result;
    }
}

File: JavaScript/Medium/BuildAnArrayWithStackOperations.js
/**
 * 1441. Build an Array With Stack Operations
 * @param {number[]} target
 * @param {number} n
 * @return {string[]}
 */
const buildArray = function (target, n) {
  let result = new Array();
  let i = 1, j = 0;
  while (i <= n) {
    if (j == target.length) break;

    if (i == target[j]) {
      result.push("Push");
      j++;
    } else {
      result.push("Push");
      result.push("Pop");
    }
    i++;
  }
  return result;
};
